internal/fleet/integration_policy/resource_test.go

```diff
@@ -75,6 +75,9 @@ func TestAccResourceIntegrationPolicy(t *testing.T) {
 func TestAccResourceIntegrationPolicySecretsFromSDK(t *testing.T) {
 	policyName := sdkacctest.RandStringFromCharSet(22, sdkacctest.CharSetAlphaNum)
 
+	sdkConstrains, err := version.NewConstraint(">=8.10.0,<8.16.0")
```

Comment:
Why do we disable it for >= 8.16.0?

Reply:
This was in the test results, prior to the selective disabling.
```
TestAccResourceIntegrationPolicySecretsFromSDK
    resource_test.go:78: Step 1/2 error: Check failed: Check 5/9 error: elasticstack_fleet_integration_policy.test_policy: Attribute 'vars_json' didn't match "{\"access_key_id\":{\"id\":\"\\S+\",\"isSecretRef\":true},\"default_region\":\"us-east-1\",\"endpoint\":\"endpoint\",\"secret_access_key\":{\"id\":\"\\S+\",\"isSecretRef\":true},\"session_token\":{\"id\":\"\\S+\",\"isSecretRef\":true}}", got "{}"
```

I think rather than constraining it to pre-8.16, just remove step 1 check 5 in the 0.11.7 check. The test passes after that, and the important part is how the data is represented in step 2.

Reply:
@daemitus , I assume step 2 is the same code as step 1 from `TestAccResourceIntegrationPolicySecrets`. My intention was to avoid code duplication and limit this case for previous versions as a migration sample (historical reference).

Would you still see the benefit of step 2 while we have  `TestAccResourceIntegrationPolicySecrets`? Please, advice, if I miss something.

Reply:
So it looks like 8.16 stopped wrapping the vars in a `{"value": "..."}` block when doing read requests.
This is actually required in the 0.11.7 provider logic, hence why we're getting an unexpected empty map (`{}`) as vars_json, rather than secretRefs. So this does indeed seem like a <8.16 legacy test. Additionally, anyone using integration policy resources on <=0.11.7 and 8.16.x will see a different kind of secret churn than they did before.

I now concur with the sdkConstrains check.

See below.

```
8.15.1: 
  "vars": {
    "shared_credential_file": { "type": "text" },
    "access_key_id": { "value": { "id": "xaJG25MBfiehiN8gM7uT", "isSecretRef": true }, "type": "password"},
    ...
  },

8.16.2
  "vars": {
    "access_key_id": { "id": "jm4l25MBNi2Twv6L31b2", "isSecretRef": true },
    ...
  },
```

Reply:
Superb, thanks. How did you managed to get debug output? somehow I wasn't able to get request / responses from integration policies by using `TF_LOG=DEBUG` 🤔 .

Reply:
@daemitus , I'm curious, kbapi is generated from Kibana `main` docs which seems to differ from the ones from 8.16. Can this lead to the test failure?

Reply:
See https://github.com/elastic/terraform-provider-elasticstack/blob/main/generated/kbapi/transform_schema.go#L959

While the api does list types for each, last time I checked the anyOf/oneOf union was broken upstream in the generated API. So instead kbapi just casts it to `any`. The current method extracts anything inside a `value` field and then continues into the regular process at https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/fleet/integration_policy/secrets.go#L83

Reply:
Thanks for the details.